features/config/dao/impl/src/main/java/org/opennms/features/config/dao/impl/util/XsdModelConverter.java

```diff
@@ -0,0 +1,385 @@
+            currentConfigItem.setType(ConfigItem.Type.OBJECT);
+            currentConfigItem.setSchemaRef("");
+            currentConfigItem.getChildren().add(child);
+        }
+    }
+
+
+    private void handleEnumerationRestrictions(ConfigItem item, HashMap<XmlSchemaRestriction.Type, List<XmlSchemaRestriction>> facets) {
+        List<XmlSchemaRestriction> enumerationFacets = facets.get(XmlSchemaRestriction.Type.ENUMERATION);
+        if (enumerationFacets != null && !enumerationFacets.isEmpty()) {
+            List<String> enumValues = enumerationFacets.stream().map(e -> (String) e.getValue()).collect(Collectors.toList());
+            item.setEnumValues(enumValues);
+        }
+    }
+
+    private void handlePatternRestrictions(ConfigItem item, HashMap<XmlSchemaRestriction.Type, List<XmlSchemaRestriction>> facets) {
+        List<XmlSchemaRestriction> patternFacets = facets.get(XmlSchemaRestriction.Type.PATTERN);
+        if (patternFacets != null && !patternFacets.isEmpty()) {
+            XmlSchemaRestriction restriction = patternFacets.get(0);
+            Object obj = restriction.getValue();
+            if (obj instanceof String) {
+                // skip apache xml walker XmlSchemaCollection.java's default pattern
+                if ((item.getType() == ConfigItem.Type.INTEGER
+                        || item.getType() == ConfigItem.Type.LONG
+                        || item.getType() == ConfigItem.Type.NUMBER
+                        || item.getType() == ConfigItem.Type.POSITIVE_INTEGER
+                        || item.getType() == ConfigItem.Type.NEGATIVE_INTEGER
+                        || item.getType() == ConfigItem.Type.NON_NEGATIVE_INTEGER) && "[\\-+]?[0-9]+".equals(obj))
+                    return;
+                else
+                    item.setPattern((String) obj);
+            }
+        }
+    }
+
+    private void handleOtherRestrictions(ConfigItem item, HashMap<XmlSchemaRestriction.Type, List<XmlSchemaRestriction>> facets) {
+        List<XmlSchemaRestriction> minFacets = facets.get(XmlSchemaRestriction.Type.INCLUSIVE_MIN);
+        if ((minFacets != null) && !minFacets.isEmpty()) {
+            // Should only be one minimum specified
+            XmlSchemaRestriction minRestriction = minFacets.get(0);
+            Object minVal = minRestriction.getValue();
+            if (minVal instanceof String) {
+                item.setMin(Long.valueOf((String) minVal));
+            }
+        }
+
+        List<XmlSchemaRestriction> maxFacets = facets.get(XmlSchemaRestriction.Type.INCLUSIVE_MAX);
+        if ((maxFacets != null) && !maxFacets.isEmpty()) {
+            // Should only be one minimum specified
+            XmlSchemaRestriction maxRestriction = maxFacets.get(0);
+            Object maxVal = maxRestriction.getValue();
+            if (maxVal instanceof String) {
+                item.setMax(Long.valueOf((String) maxVal));
+            }
+        }
+
+        // openapi do not support exclusive
```

Comment:
Are you sure? https://swagger.io/docs/specification/data-models/data-types/#numbers

"To exclude the boundary values, specify exclusiveMinimum: true and exclusiveMaximum: true. "